lib/swagger/swagger.ex

```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
-  defp has_key?({:ok, map}, key), do: Map.has_key?(map, key)
-
-  defp handle_request(true, swagger_paths, request), do: add_response(swagger_paths, request)
-  defp handle_request(false, _swagger_paths, request), do: format_request(request)
-
-  defp add_response(swagger_map, request) do
-    original_request = swagger_map[request.path][request.verb]
+  defp build_path_item_object(%{verb: verb} = request),
+    do: %{verb => build_operation_object(request)}
 
+  defp build_operation_object(request) do
     %{
-      request.verb =>
-        Map.put(
-          original_request,
-          "responses",
-          Map.merge(original_request["responses"], Formatter.format_responses(request))
-        )
+      "summary" => Descriptor.get_action_description(request),
+      "description" => Descriptor.get_request_description(request),
+      "responses" => Formatter.format_responses(request)
     }
+    |> parameters_if_needed(request)
+    |> request_body_if_needed(request)
+    |> security_if_needed(request)
   end
 
-  defp format_request(request) do
-    operation =
-      %{
-        "summary" => Descriptor.get_action_description(request),
-        "description" => Descriptor.get_request_description(request),
-        "responses" => Formatter.format_responses(request)
-      }
-      |> put_parameters_if_needed(request)
-      |> put_request_body_if_needed(request)
-      |> put_security_if_needed(request)
+  defp include_security_scheme(openapi_object, requests) do
+    requests
+    |> Enum.any?(&has_authorization_header?/1)
+    |> put_security_scheme(openapi_object)
+  end
 
-    %{
-      request.verb => operation
-    }
+  defp put_security_scheme(false, openapi_object), do: openapi_object
+
+  defp put_security_scheme(true, openapi_object),
+    do: Map.put(openapi_object, "components", @security_scheme)
+
+  defp has_authorization_header?(request) do
+    request
+    |> Map.fetch!(:header_params)
+    |> Enum.any?(fn {h, _} -> String.match?(h, ~r/^authorization$/i) end)
   end
 
-  defp put_parameters_if_needed(
-         swagger,
-         %{path_params: params, query_params: query, header_params: headers} = request
-       )
-       when params not in [nil, %{}] or query not in [nil, %{}] or headers not in [nil, []] do
-    Map.put(swagger, "parameters", Formatter.format_parameters(request))
+  defp parameters_if_needed(openapi_object, request) do
+    request
+    |> Map.take([:path_params, :query_params, :header_params])
+    |> Enum.any?(&(elem(&1, 1) not in @empty_data))
+    |> include_parameters(openapi_object, request)
   end
 
-  defp put_parameters_if_needed(swagger, _), do: swagger
+  defp include_parameters(false, openapi_object, _request), do: openapi_object
 
-  defp put_request_body_if_needed(swagger, %{request_body: body} = request)
-       when body not in [nil, %{}] do
-    Map.put(swagger, "requestBody", Formatter.format_body(request))
-  end
+  defp include_parameters(true, openapi_object, request),
```

Comment:
You have a break line here. You can remove the inline notation

```elixir
defp include_parameters(true, openapi_object, request) do
  Map.put(openapi_object, "parameters", Formatter.request_parameters(request))
end
```

Reply:
:+1: